hotshot-task-impls/src/quorum_vote/handlers.rs

```diff
@@ -390,21 +391,44 @@ pub(crate) async fn handle_quorum_proposal_validated<
         .await
     };
 
+    if let Some(cert) = &task_state.staged_epoch_upgrade_certificate {
+        if leaf_views.last().unwrap().leaf.height() >= task_state.epoch_upgrade_block_height {
```

Comment:
So here we are waiting to apply the upgrade until the configured height, but we already set the first epoch, I think that needs to happen here